praw/models/reddit/submission.py

```diff
@@ -905,6 +870,52 @@ def hide(self, *, other_submissions: list[praw.models.Submission] | None = None)
         ):
             self._reddit.post(API_PATH["hide"], data={"id": submissions})
 
+    def id_from_url(self, url: str) -> str:
+        """Return the ID contained within a submission URL.
+
+        :param url: A url to a submission in one of the following formats (http urls
+            will also work):
+
+            - ``"https://redd.it/2gmzqe"``
+            - ``"https://reddit.com/comments/2gmzqe/"``
+            - ``"https://www.reddit.com/r/redditdev/comments/2gmzqe/praw_https/"``
+            - ``"https://www.reddit.com/gallery/2gmzqe"``
+            - ``"https://www.reddit.com/r/space/s/skZgKbbMDq"``
+
+        :raises: :class:`.InvalidURL` if ``url`` is not a valid submission URL.
+
+        """
+        parts = self._url_parts(url)
+
+        if "s" in parts:  # handling new share urls from mobile apps
```

Comment:
Can we extract this logic into a single location since it's the same for both submissions and comments?